syntax = "proto3";

package silenteight.datasource.api.event;

import "google/api/annotations.proto";
import "silenteight/datasource/api/common/common.proto";

option java_package = "com.silenteight.datasource.api.event";
option java_outer_classname = "EventInputProto";
option java_multiple_files = true;

/**
  Intended for Date logical discounting
 */
service EventInputService {
  rpc BatchGetMatchEventInputsV1 (silenteight.datasource.api.common.BatchGetMatchAgentInputsRequest) returns (stream BatchGetMatchEventInputsV1Response) {
    option (google.api.http) = {
      post: "/agent-inputs/event/versions/1/matches:batchGet",
      body: "*"
    };
  }
}

message BatchGetMatchEventInputsV1Response {
  repeated EventInput EventInputs = 1;
}

message EventInput {
  string match = 1;
  repeated EventFeatureInput event_feature_inputs = 2;
}

message EventFeatureInput {
  string feature = 1;
  repeated string alerted_party_dates = 2;
  repeated string watchlist_events = 3;
}
